### PR TITLE
Cp scope 4 dependencies from recorded activities

### PR DESCRIPTION
**Pull request by luke-moth**

### CODE REVIEW

**Comment by luke-moth** [comment]
reminder for myself
```python

def temp_visualisation():
    from openclsim.plot.vessel_planning import get_gantt_chart
    import plotly.graph_objs as go
    my_cp = DependenciesFromSimpy(**simulation_2_barges_custom_env)
    critical_df = my_cp.get_critical_path_df()
    data = get_gantt_chart(simulation_2_barges_custom_env['object_list'],
                           id_map=simulation_2_barges_custom_env['activity_list'],
                           static=True, critical_path_dataframe=critical_df)
    fig = go.Figure(**data)
    fig.show()
    fig.write_html(r"D:\temp_files\crit1_2barges.html")

    my_cp = DependenciesFromSimpy(**simulation_4_barges_custom_env)
    critical_df = my_cp.get_critical_path_df()
    data = get_gantt_chart(simulation_4_barges_custom_env['object_list'],
                           id_map=simulation_4_barges_custom_env['activity_list'],
                           static=True, critical_path_dataframe=critical_df)
    fig = go.Figure(**data)
    fig.show()
    fig.write_html(r"D:\temp_files\crit1_4barges.html")


    my_cp = DependenciesFromSimpy(**simulation_simple)
    critical_df = my_cp.get_critical_path_df()
    data = get_gantt_chart(simulation_simple['object_list'],
                           id_map=simulation_simple['activity_list'],
                           static=True, critical_path_dataframe=critical_df)
    fig = go.Figure(**data)
    fig.show()
    fig.write_html(r"D:\temp_files\simple_while.html")
```

File: src/openclsim/critical_path/dependencies_from_recorded_activities.py [hunk 1]
```diff
@@ -16,16 +16,74 @@ def get_dependency_list(self):
         """
         Get dependencies directly from recorded activities (match on identical timestamps)
```
Comment (FKSWB) [inline src/openclsim/critical_path/dependencies_from_recorded_activities.py:17]: .

Comment (FKSWB) [reply src/openclsim/critical_path/dependencies_from_recorded_activities.py:17]: (add period at end of sentence)

File: src/openclsim/critical_path/dependencies_from_recorded_activities.py [hunk 2]
```diff
@@ -16,16 +16,74 @@ def get_dependency_list(self):
...
-        # ...
-        return []
+        if self.dependency_list is None:
+            self.__set_dependency_list()
+
+        return self.dependency_list
+
+    def __set_dependency_list(self):
+        """
+        Hidden and protected method for the get_dependency_list.
```
Comment (FKSWB) [inline src/openclsim/critical_path/dependencies_from_recorded_activities.py:57]: should be single line by convention?

Comment (FKSWB) [reply src/openclsim/critical_path/dependencies_from_recorded_activities.py:57]: i.e. single line as """{str}"""

File: tests/test_critical_dependencies_from_recorded_activities.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
+"""
+Tests for openclsim.critical_path.dependencies_from_recorded_activities module
```
Comment (FKSWB) [inline tests/test_critical_dependencies_from_recorded_activities.py:2]: .

File: tests/test_critical_dependencies_from_recorded_activities.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    assert hasattr(my_cp, "get_dependency_list") and callable(
+        getattr(my_cp, "get_dependency_list")
+    ), "this method 'get_dependency_list' should exist!"
+    assert issubclass(
+        DependenciesFromRecordedActivities, BaseCP
+    ), "should be child of BaseCP"
+
+
+def test_get_dependency_list(simulation_2_barges):
+    """test get dependency list"""
```
Comment (FKSWB) [inline tests/test_critical_dependencies_from_recorded_activities.py:22]: capital and .

File: tests/test_critical_dependencies_from_recorded_activities.py [hunk 5]
```diff
@@ -0,0 +1,30 @@
+"""
```
Comment (FKSWB) [inline tests/test_critical_dependencies_from_recorded_activities.py:1]: General: do the assertion error messages live up to the standard of OpenCLSim?

Comment (luke-moth) [reply tests/test_critical_dependencies_from_recorded_activities.py:1]: I assume assert in pytest is accepted/desirable...

**Review by Pietervanhalem — APPROVED** [review]

**Pull request merged**